src/persist/src/operators/stream.rs

```diff
@@ -226,21 +226,31 @@ where
                         !PartialOrder::less_equal(&new_input_frontier, &input_frontier.borrow());
 
                     if progress {
-                        input_frontier.clear();
-                        input_frontier.extend(new_input_frontier.into_iter().cloned());
-
                         // Only try and seal if we have seen some data. Otherwise, we wouldn't have
                         // a capability that allows us to emit errors.
                         if let Some(err_cap) = capabilities.get(0) {
-                            if let Some(data_frontier) = input_frontier.elements().first() {
-                                log::trace!("Sealing {} up to {}", &operator_name, data_frontier);
+                            for frontier_element in new_input_frontier.iter() {
+                                // Only seal if this element of the new input frontier truly
```

Comment:
Could this instead subsume the `if progress` check?

Reply:
I'm also managing the capabilities and updating the input frontier only if there is progress. So I think we still need that in addition to checking each frontier element.